src/main/java/apoc/export/cypher/formatter/AbstractCypherFormatter.java

```diff
@@ -127,18 +129,23 @@ public void buildStatementForNodes(String nodeClause, String setClause,
 				// properties
 				out.append(", ");
 				out.append("properties:");
-				writeProperties(out, props, (es) -> !idMap.containsKey(es.getKey()));
+
 				propertiesCount.addAndGet(props.size());
+				props = props.entrySet()
+						.stream()
+						.filter(es -> !idMap.containsKey(es.getKey()))
+						.collect(Collectors.toMap(es -> es.getKey(), es -> es.getValue()));
+				writeProperties(out, props);
 
 				// end element
 				out.append("}");
 				boolean isEnd = last.equals(node);
 				if (isEnd || batchCount.get() % unwindBatchSize == 0 || batchCount.get() % batchSize == 0) {
-					int unwindLength = (i + 1) % unwindBatchSize;
-					int unwindLeftOver = (unwindBatchSize - unwindLength) % unwindBatchSize;
-					batchCount.addAndGet(unwindLeftOver);
 					writeUnwindEnd(exportConfig, out, batchCount, isEnd);
 					if(isEnd || writeClosingStatements(exportConfig, batchCount)) {
+						int unwindLength = (i + 1) % unwindBatchSize;
```

Comment:
can't we simplify this more to a simple chunking?
where we chunk on the smaller of batchSize/unwindSIze